sdk/identity/azure-identity/src/main/java/com/azure/identity/AadCredentialBuilderBase.java

```diff
@@ -70,4 +70,31 @@ public T executorService(ExecutorService executorService) {
         this.identityClientOptions.setExecutorService(executorService);
         return (T) this;
     }
+
+    /**
+     * Sets whether to use an unprotected file specified by <code>cacheFileLocation()</code> instead of
+     * Gnome keyring on Linux. This is false by default.
+     *
+     * @param useUnprotectedFileOnLinux whether to use an unprotected file for cache storage.
+     *
+     * @return An updated instance of this builder with the unprotected token cache setting set as specified.
+     */
+    @SuppressWarnings("unchecked")
+    public T useUnprotectedTokenCacheFileOnLinux(boolean useUnprotectedFileOnLinux) {
+        this.identityClientOptions.setUseUnprotectedTokenCacheFileOnLinux(useUnprotectedFileOnLinux);
+        return (T) this;
+    }
+
+    /**
+     * Disable using the shared token cache.
+     *
+     * @param disabled whether to disable using the shared token cache.
+     *
+     * @return An updated instance of this builder with if the shared token cache disabled specified.
+     */
+    @SuppressWarnings("unchecked")
+    public T disableSharedTokenCache(boolean disabled) {
```

Comment:
I think that persistence of the token cache should be opt-in rather than opt out. Also I'm not sure we can put this on the base builder for all credentials, for instance what does it mean to create a `SharedTokenCachceCredential` with the shared token cache disabled.

Reply:
For SharedTokenCacheCredential, yes you can disable shared token cache, but assuming no one will do this does save us a ton of duplicate code.

Reply:
ManagedIdentityCredential and DefaultAzureCredential inherit from CredentialBuilderBase, so they will not inherit this setter currently.